### PR TITLE
[Mercurial] profits beneficiary

### PR DESCRIPTION
**Pull request by cnek**
much similar to https://github.com/UXDProtocol/uxd-program/pull/221
except: 
- profits_beneficiary_collateral's owner compare directly with depository.profits_beneficiary_key, without passing the profits_beneficiary_key to context
- require non zero pubkey as profits_beneficiary_key when calling collect profit ix, abort otherwise
- added test for above scenraios

client PR: https://github.com/UXDProtocol/uxd-client/pull/37

### CODE REVIEW

File: programs/uxd/src/error.rs [hunk 1]
```diff
@@ -74,7 +74,11 @@ pub enum UxdError {
     InvalidDepositoryCollateral,
     #[msg("The provided depository shares does not match the depository's one.")]
     InvalidDepositoryShares,
-    #[msg("The provided mercurial vault does not match the Depository's one.")]
+    #[msg("The Profits beneficiary provided does not match the depository's one.")]
+    InvalidProfitsBeneficiary,
+    #[msg("The Profits beneficiary provided does not match the depository's one.")]
+    ProfitsBeneficiaryNotInitialized,
+    #[msg("The profits beneficiary hasn't setup for this depository")]
```
Comment (acamill) [inline programs/uxd/src/error.rs:81]: The profits beneficiary hasn't setup for this depository
> 
The profits beneficiary hasn't been setup for this depository

File: programs/uxd/src/events.rs [hunk 2]
```diff
@@ -96,6 +96,19 @@ pub struct SetDepositoryMintingDisabledEvent {
...
+#[event]
+pub struct SetDepositoryProfitsBeneficiaryKeyEvent {
+    #[index]
+    pub version: u8,
+    #[index]
+    pub controller: Pubkey,
+    #[index]
+    pub depository: Pubkey,
+    #[index]
+    pub profits_beneficiary_key: Pubkey,
```
Comment (acamill) [inline programs/uxd/src/events.rs:109]: I wouldn't call that key as we don't use that nomenclature anywhere else

File: tests/suite/editMercurialVaultDepositorySuite.ts [hunk 3]
```diff
@@ -107,13 +107,29 @@ export const editMercurialVaultDepositorySuite = async function ({
       });
     });
 
+    it(`Edit profitsBeneficiaryKey alone should work`, async function () {
```
Comment (uxd-vincent) [inline tests/suite/editMercurialVaultDepositorySuite.ts:110]: I applied your comment on my branch, we can simply use: "profitsBeneficiaryCollateral" everywhere instead now :)

File: programs/uxd/src/error.rs (hunk 1 above)
Comment (cnek) [reply programs/uxd/src/error.rs:81]: grammar 🥲

File: programs/uxd/src/events.rs (hunk 2 above)
Comment (cnek) [reply programs/uxd/src/events.rs:109]: yes, left the same comments on the crefix PR, was for alignment for the naming,
but gonna get rid of this field now but just `profits_beneficiary_collateral` as discussed

**Comment by cnek** [comment]
aligned much with #221 now, would merge and continue the work on #214

**Pull request merged**